In [64]:
from pyspark.sql import SparkSession
from config import db_params
from pyspark.sql.functions import mean,count,explode,col,monotonically_increasing_id,lit

spark = SparkSession.builder \
    .appName("Manchester_UTD Analysis")\
    .config("spark.jars", "/home/jss/app/spark/spark-3.5.0-bin-hadoop3/jars/postgresql-42.7.2.jar")\
    .getOrCreate()

In [65]:
jdbc_url=db_params["jdbc_url"]
user=db_params["user"]
password=db_params["password"]
driver=db_params["driver"]

In [66]:
table_name = "teams"
team_df = spark.read \
    .format("jdbc") \
    .option("url", jdbc_url) \
    .option("dbtable", table_name) \
    .option("user", user) \
    .option("password", password) \
    .option("driver",driver) \
    .load()

In [67]:
table_name = "players"
player_df = spark.read \
    .format("jdbc") \
    .option("url", jdbc_url) \
    .option("dbtable", table_name) \
    .option("user", user) \
    .option("password", password) \
    .option("driver",driver) \
    .load()

In [68]:
table_name = "coachs"
coach_df = spark.read \
    .format("jdbc") \
    .option("url", jdbc_url) \
    .option("dbtable", table_name) \
    .option("user", user) \
    .option("password", password) \
    .option("driver",driver) \
    .load()

In [69]:
table_name = "venues"
venues_df = spark.read \
    .format("jdbc") \
    .option("url", jdbc_url) \
    .option("dbtable", table_name) \
    .option("user", user) \
    .option("password", password) \
    .option("driver",driver) \
    .load()

In [70]:
table_name = "leagues"
leagues_df = spark.read \
    .format("jdbc") \
    .option("url", jdbc_url) \
    .option("dbtable", table_name) \
    .option("user", user) \
    .option("password", password) \
    .option("driver",driver) \
    .load()

In [71]:
table_name = "countries"
countries_df = spark.read \
    .format("jdbc") \
    .option("url", jdbc_url) \
    .option("dbtable", table_name) \
    .option("user", user) \
    .option("password", password) \
    .option("driver",driver) \
    .load()

In [72]:
england_fixture_statistics = spark.read.parquet("file:///home/jss/football_parquet/datas/england/fixtures/statistics/*.parquet", multiLine=True)
england_fixture_events = spark.read.parquet("file:///home/jss/football_parquet/datas/england/fixtures/events/*.parquet", multiLine=True)
england_player_stat = spark.read.parquet("file:///home/jss/football_parquet/datas/england/playerstat/*.parquet", multiLine=True)
england_team_stat = spark.read.parquet("file:///home/jss/football_parquet/datas/england/teamstat/*.parquet", multiLine=True)

In [73]:
spain_fixture_statistics = spark.read.parquet("file:///home/jss/football_parquet/datas/spain/fixtures/statistics/*.parquet", multiLine=True)
spain_fixture_events = spark.read.parquet("file:///home/jss/football_parquet/datas/spain/fixtures/events/*.parquet", multiLine=True)
spain_player_stat = spark.read.parquet("file:///home/jss/football_parquet/datas/spain/playerstat/*.parquet", multiLine=True)
spain_team_stat = spark.read.parquet("file:///home/jss/football_parquet/datas/spain/teamstat/*.parquet", multiLine=True)

In [74]:
germany_fixture_statistics = spark.read.parquet("file:///home/jss/football_parquet/datas/germany/fixtures/statistics/*.parquet", multiLine=True)
germany_fixture_events = spark.read.parquet("file:///home/jss/football_parquet/datas/germany/fixtures/events/*.parquet", multiLine=True)
germany_player_stat = spark.read.parquet("file:///home/jss/football_parquet/datas/germany/playerstat/*.parquet", multiLine=True)
germany_team_stat = spark.read.parquet("file:///home/jss/football_parquet/datas/germany/teamstat/*.parquet", multiLine=True)

In [75]:
italy_fixture_statistics = spark.read.parquet("file:///home/jss/football_parquet/datas/italy/fixtures/statistics/*.parquet", multiLine=True)
italy_fixture_events = spark.read.parquet("file:///home/jss/football_parquet/datas/italy/fixtures/events/*.parquet", multiLine=True)
italy_player_stat = spark.read.parquet("file:///home/jss/football_parquet/datas/italy/playerstat/*.parquet", multiLine=True)
italy_team_stat = spark.read.parquet("file:///home/jss/football_parquet/datas/italy/teamstat/*.parquet", multiLine=True)

In [76]:
france_fixture_statistics = spark.read.parquet("file:///home/jss/football_parquet/datas/france/fixtures/statistics/*.parquet", multiLine=True)
france_fixture_events = spark.read.parquet("file:///home/jss/football_parquet/datas/france/fixtures/events/*.parquet", multiLine=True)
france_player_stat = spark.read.parquet("file:///home/jss/football_parquet/datas/france/playerstat/*.parquet", multiLine=True)
france_team_stat = spark.read.parquet("file:///home/jss/football_parquet/datas/france/teamstat/*.parquet", multiLine=True)

In [77]:
player_sideline = spark.read.parquet("file:///home/jss/football_parquet/datas/sidelined/player/*.parquet", multiLine=True)
coach_sideline = spark.read.parquet("file:///home/jss/football_parquet/datas/sidelined/coach/*.parquet", multiLine=True)

In [78]:
england_fixture_statistics.show()

+----------+-------+---------------+-------------+------------+-----+----------------+--------+--------+---------------+---------+---------------+--------------+-------------+----------------+-----------+------------+------------+--------------+
|fixture_id|team_id|Ball Possession|Blocked Shots|Corner Kicks|Fouls|Goalkeeper Saves|Offsides|Passes %|Passes accurate|Red Cards|Shots insidebox|Shots off Goal|Shots on Goal|Shots outsidebox|Total Shots|Total passes|Yellow Cards|expected_goals|
+----------+-------+---------------+-------------+------------+-----+----------------+--------+--------+---------------+---------+---------------+--------------+-------------+----------------+-----------+------------+------------+--------------+
|   1035117|     35|            34%|            2|           3|   15|               5|       0|     75%|            228|        1|              4|             1|            4|               3|          7|         303|           2|          0.63|
|   1035117|    

In [79]:
england_player_stat.show()

+---------+-------+---------+-------------+----------------+------------+------------+-----------+-------------+-----------+------------+------+-------+---------+-----------+--------+-----------+--------------+-------------+-----------+------------+----------+---------------+-------------+--------------+---------------------+-----------+---------+-----------------+----------------+-------------+-----------+---------------+------------+---------------+---------+-----------+----------------+--------------+--------------+-------------+
|player_id|team_id|league_id|league_season|game_appearances|game_lineups|game_minutes|game_number|game_position|game_rating|game_captain|sub_in|sub_out|sub_bench|total_shots|shots_on|total_goals|goals_conceded|goals_assists|goals_saves|passes_total|passes_key|passes_accuracy|tackles_total|tackles_blocks|tackles_interceptions|duels_total|duels_won|dribbles_attempts|dribbles_success|dribbles_past|fouls_drawn|fouls_committed|cards_yellow|cards_yellowred|cards_

In [80]:
england_team_stat.createOrReplaceTempView("england_team_stat")

In [81]:
england_team_stat.show()

+------------+---------+------+-------+--------------------+---------+---------+----------+---------+---------+----------+----------+----------+-----------+----------+----------+-----------+----------+----------+-----------+--------------+--------------+---------------+----------------+---------------------+-----------------+----------------------+-----------------+----------------------+-----------------+----------------------+-----------------+----------------------+-----------------+----------------------+------------------+-----------------------+-------------------+------------------------+-----------------------+-------------------------+------------------------+-------------------------+------------------------+-------------------------+------------------------+-------------------------+------------------------+-------------------------+------------------------+--------------------------+-------------------------+---------------------------+--------------------------+-----------

In [82]:
england_player_stat.createOrReplaceTempView("england_player_stat")

In [83]:
england_fixture_statistics.createOrReplaceTempView("englad_fixture_statistics")

In [84]:
spain_team_stat.createOrReplaceTempView("spain_team_stat")
spain_player_stat.createOrReplaceTempView("spain_player_stat")
spain_fixture_statistics.createOrReplaceTempView("spain_fixture_statistics")

In [85]:
germany_team_stat.createOrReplaceTempView("germany_team_stat")
germany_player_stat.createOrReplaceTempView("germany_player_stat")
germany_fixture_statistics.createOrReplaceTempView("germany_fixture_statistics")

In [86]:
italy_team_stat.createOrReplaceTempView("italy_team_stat")
italy_player_stat.createOrReplaceTempView("italy_player_stat")
italy_fixture_statistics.createOrReplaceTempView("italy_fixture_statistics")

In [87]:
france_team_stat.createOrReplaceTempView("france_team_stat")
france_player_stat.createOrReplaceTempView("france_player_stat")
france_fixture_statistics.createOrReplaceTempView("france_fixture_statistics")

In [88]:
team_df.createOrReplaceTempView("team_df")

In [89]:
team_df.show()

+-------+-----------+---------+------------+--------------------+------------+
|team_id|  team_name|team_code|team_founded|           team_logo|country_name|
+-------+-----------+---------+------------+--------------------+------------+
|    498|  Sampdoria|      SAM|        1946|https://media.api...|       Italy|
|    507|     Ascoli|      ASC|        1898|https://media.api...|       Italy|
|    508|       Bari|      BAR|        1908|https://media.api...|       Italy|
|    510| Cittadella|      CIT|        1973|https://media.api...|       Italy|
|    515|     Spezia|      SPE|        1906|https://media.api...|       Italy|
|    516|    Ternana|      TER|        1925|https://media.api...|       Italy|
|    517|    Venezia|      VEN|        1907|https://media.api...|       Italy|
|    518|    Brescia|      BRE|        1911|https://media.api...|       Italy|
|    520|  Cremonese|      CRE|        1903|https://media.api...|       Italy|
|    522|    Palermo|      PAL|        1900|https://

In [90]:
player_df.createOrReplaceTempView("player_df")

In [91]:
player_df.show()

+---------+--------------------+---------------+------------+-------------+-------------+--------------------+------------------+-----------------+-------------------+
|player_id|    player_firstname|player_lastname|player_birth|player_height|player_weight|        player_photo|player_nationality|     player_place|player_placecountry|
+---------+--------------------+---------------+------------+-------------+-------------+--------------------+------------------+-----------------+-------------------+
|       18|         Jadon Malik|         Sancho|  2000-03-25|          180|           76|https://media.api...|           England|           London|            England|
|       25|               Marco|           Reus|  1989-05-31|          180|           71|https://media.api...|           Germany|         Dortmund|            Germany|
|       26|              Marius|           Wolf|  1995-05-27|          188|           81|https://media.api...|           Germany|           Coburg|            G

In [92]:
player_sideline.createOrReplaceTempView("player_sideline")

In [93]:
player_sideline.show()

+---------+----------------+--------------+------------+
|player_id|   sideline_type|sideline_start|sideline_end|
+---------+----------------+--------------+------------+
|       18|    Groin Injury|    2024-01-31|  2024-02-08|
|       18|       Suspended|    2023-09-14|  2024-01-11|
|       18|         Illness|    2023-02-27|  2023-03-04|
|       18|         Illness|    2022-10-31|  2022-11-14|
|       18|         Illness|    2022-07-24|  2022-08-06|
|       18|         Illness|    2022-04-27|  2022-06-30|
|       18|Hip/Thigh Injury|    2022-02-06|  2022-02-08|
|       18|    Thigh Injury|    2022-01-04|  2022-01-14|
|       18|    Thigh Injury|    2021-09-04|  2021-09-10|
|       18|           Knock|    2021-08-13|  2021-08-13|
|       18|    Thigh Injury|    2021-03-03|  2021-04-20|
|       18|       Pneumonia|    2020-09-30|  2020-10-10|
|       18|           Knock|    2019-12-18|  2019-12-19|
|       18|           Knock|    2019-11-06|  2019-11-08|
|       18|           Knock|   

In [94]:
manu_stat=spark.sql("select tf.team_name,es.* from england_team_stat es join team_df tf on es.team_id = tf.team_id where team_name like 'Manchester United'")

In [95]:
manu_stat.show()

+-----------------+------------+---------+------+-------+--------------------+---------+---------+----------+---------+---------+----------+----------+----------+-----------+----------+----------+-----------+----------+----------+-----------+--------------+--------------+---------------+----------------+---------------------+-----------------+----------------------+-----------------+----------------------+-----------------+----------------------+-----------------+----------------------+-----------------+----------------------+------------------+-----------------------+-------------------+------------------------+-----------------------+-------------------------+------------------------+-------------------------+------------------------+-------------------------+------------------------+-------------------------+------------------------+-------------------------+------------------------+--------------------------+-------------------------+---------------------------+--------------------

In [96]:
total_against = spark.sql("""SELECT (
    `16-30_total_against_goals` +
    `31-45_total_against_goals` +
    `46-60_total_against_goals` +
    `61-75_total_against_goals` +
    `76-90_total_against_goals` +
    `91-105_total_against_goals`
) as total_against
FROM england_team_stat
WHERE team_id = 33;
""")

In [97]:
total_against.show()

+-------------+
|total_against|
+-------------+
|           29|
+-------------+



In [98]:
total_yellow = spark.sql("SELECT (`0-15_yellow`+`16-30_yellow`+`31-45_yellow`+`46-60_yellow`+`61-75_yellow`+`76-90_yellow`+`91-105_yellow`) AS total_yellow,(`0-15_yellow`+`16-30_yellow`+`31-45_yellow`+`46-60_yellow`+`61-75_yellow`+`76-90_yellow`+`91-105_yellow`)/total_game as average_yellow  FROM england_team_stat where team_id = 33" )

In [99]:
total_yellow.show()

+------------+--------------+
|total_yellow|average_yellow|
+------------+--------------+
|          62|          2.48|
+------------+--------------+



- 25게임 기준 전체 14승 홈 원정 7게임 승리했지만 2경기 무승부 9경기 패배 홈에서는 4번 원정에서는 5번을 패배를 했다.
- 무승부가 적다는것은  이길게임을 이기거나 비길게임을 이기거나 지거나 패배할 게임은 그대로 패배를 했다는 이야기다.
- 홈 득점은 18득점 원정 골은 17득점 총 득점은 35득점 홈 평균 득점은 경기당 1.5 원정 평균득점 1.3 총 평균득점 1.4득점
- 16-30분대에 7골로 19.44% 76-90분대 8골 22.22% 득점을 많이 많이했다
- 25경기 기준 34실점을 하고 전반 0-15분대에 5골을 실점을 했다
- 16-30분대에도 8골로 24.24%로 실점을 하고 61-75분대에 7골로 21.21% 실점이 많았다
- 추가시간대인 91-105분대에 4득점 총득점의 11.11% 3실점 총 실점의 9.09%를 차지한다
- 클린시트 홈에서 3경기 원정에서 4경기 총 7경기를 했다
- 무득점 홈에서 3경기 원정에서 4경기 총 7경기를 했다
- 62장의 yellow card를 받고 경기 막판 76분-90분대와 91분대에서 105분대에 yellow card의 큰 비중을 갖고있다
- 경기당 평균 2.48의 옐로우카드를 받았다
- 퇴장 추가시간대인 91-105경기에 한개 받았다

# 문제
- 실점도 아주 적은편은 아니지만 제일 심각한 문제는 득점이 너무 낮다
- 득점력을 높여야 질 수도 있는 게임을 비기게 해줌으로써 상위권 팀들과 경쟁을 할수 있다
- 득점이 많은 76-90분보다 이른 61-75분대까지의 실점을 줄여야한다 
- 옐로우 카드를 많이 받는 시간이 경기 막판인데 경기 막판에 급해지는 경향때문에 카드를 많이 받는다

In [100]:
manu_player_goalkeeper_stat = spark.sql("""select distinct tf.team_name,pf.player_firstname,pf.player_lastname,pf.player_birth,pf.player_nationality,TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age,ps.* from england_player_stat ps join team_df tf on ps.team_id = tf.team_id join player_df pf on ps.player_id = pf.player_id where team_name like 'Manchester United' and game_position like 'Goalkeeper' """)

In [101]:
manu_player_goalkeeper_stat.show()

+-----------------+----------------+---------------+------------+------------------+----------+---------+-------+---------+-------------+----------------+------------+------------+-----------+-------------+-----------+------------+------+-------+---------+-----------+--------+-----------+--------------+-------------+-----------+------------+----------+---------------+-------------+--------------+---------------------+-----------+---------+-----------------+----------------+-------------+-----------+---------------+------------+---------------+---------+-----------+----------------+--------------+--------------+-------------+
|        team_name|player_firstname|player_lastname|player_birth|player_nationality|player_age|player_id|team_id|league_id|league_season|game_appearances|game_lineups|game_minutes|game_number|game_position|game_rating|game_captain|sub_in|sub_out|sub_bench|total_shots|shots_on|total_goals|goals_conceded|goals_assists|goals_saves|passes_total|passes_key|passes_accura

In [102]:
manu_player_goalkeeper_avg_saves_passes = spark.sql("select distinct player_id,goals_saves/game_appearances as avg_saves, passes_total/game_appearances as avg_passes,duels_won/duels_total duel_succeed_percentage from england_player_stat where team_id = 33 and game_position like 'Goalkeeper'")

In [103]:
manu_player_goalkeeper_avg_saves_passes.show()

+---------+---------+----------+-----------------------+
|player_id|avg_saves|avg_passes|duel_succeed_percentage|
+---------+---------+----------+-----------------------+
|      526|      3.6|     35.68|     0.8666666666666667|
|    19088|     NULL|      NULL|                    1.0|
|   284382|     NULL|      NULL|                   NULL|
|   138804|     NULL|      NULL|                   NULL|
|      882|     NULL|      NULL|                   NULL|
|    20319|     NULL|      NULL|                   NULL|
|    50132|     NULL|      NULL|                   NULL|
|     2931|     NULL|      NULL|                   NULL|
|   378981|     NULL|      NULL|                   NULL|
|   284361|     NULL|      NULL|                   NULL|
|     2930|     NULL|      NULL|                   NULL|
+---------+---------+----------+-----------------------+



## 골키퍼 부분
- 골키퍼 포지션에서는 오나나만 출전했다
- 게임 평점은 7.18로 높은 편이다
- 25경기 기준 총 골 세이브는 90 으로 경기당 3.6 세이브를 한다
- 25경기 기준 총 892개의 패스를 하고 경기당 35.68의 패스를 한다
- 패스 정확도 25%를 기록하고 있다
- 경합 승률 약 87%에 육박한다

## 결론
- 맨체스터 유나이티드의 골키퍼는 오나나 골키퍼만 사용하는데 평점도 높고 세이브도 많이 하는 편이다
- 실점이 적은것은 아니지만 골키퍼만 수비를 하는게 아니기 때문에 세이브도 많고 경합 승률도 높다
- 단점으로는 패스의 정확도가 너무 낮다 
- 오나나는 맨체스터 유나이티드에서 무난한 활약을 하고있다
- yellow카드 4개를 받고 있어서 1장을 더받아 결장을 할시 후보 골키퍼의 경기를 알수 없기때문에 예측할수없는 일이 발생할경우 골키퍼가 흔들릴수 있다
- 주전 골키퍼를 새로 영입할 필요는 없다

In [104]:
manu_player_defender_stat = spark.sql("""SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname,pf.player_birth,pf.player_nationality,TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, ps.*, AVG(ps.game_rating) OVER () as avg_game_rating,duels_won/duels_total,dribbles_success/dribbles_attempts,game_lineups/game_appearances,passes_total/(game_minutes/90),passes_key/(game_minutes/90),tackles_blocks/(game_minutes/90),tackles_interceptions/(game_minutes/90),duels_total/(game_minutes/90),dribbles_attempts/(game_minutes/90),game_minutes/90 total_games,total_shots/(game_minutes/90),shots_on/(game_minutes/90),goals_assists/(game_minutes/90),passes_key/passes_total,shots_on/total_shots,goals_assists/passes_key,fouls_drawn/(game_minutes/90),fouls_committed/(game_minutes/90),cards_yellow/(game_minutes/90),cards_yellow/fouls_committed,cards_yellowred/(game_minutes/90),cards_red/fouls_committed,total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
FROM england_player_stat ps
JOIN team_df tf ON ps.team_id = tf.team_id
JOIN player_df pf ON ps.player_id = pf.player_id
WHERE tf.team_name LIKE 'Manchester United' AND ps.game_position LIKE 'Defender' AND ps.game_appearances > 1;

""")

In [105]:
manu_player_defender_stat.show(n=100, truncate=False)

24/03/12 16:04:06 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 16:04:06 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 16:04:06 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 16:04:07 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 16:04:07 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 16:04:07 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 1

+-----------------+----------------+------------------+------------+------------------+----------+---------+-------+---------+-------------+----------------+------------+------------+-----------+-------------+-----------+------------+------+-------+---------+-----------+--------+-----------+--------------+-------------+-----------+------------+----------+---------------+-------------+--------------+---------------------+-----------+---------+-----------------+----------------+-------------+-----------+---------------+------------+---------------+---------+-----------+----------------+--------------+--------------+-------------+-----------------+-------------------------+--------------------------------------+---------------------------------+------------------------------------+----------------------------------+--------------------------------------+---------------------------------------------+-----------------------------------+-----------------------------------------+-------------

In [106]:
manu_defender_sidesline = spark.sql("""SELECT ps.player_id, pd.player_firstname, pd.player_lastname,COUNT(ps.player_id) as player_count
FROM england_player_stat p
JOIN player_df pd ON p.player_id = pd.player_id
JOIN player_sideline ps ON p.player_id = ps.player_id
WHERE p.team_id = 33 AND p.game_position LIKE 'Defender' AND p.game_appearances > 1 AND ps.sideline_start > '2023-08-10'
GROUP BY ps.player_id, pd.player_firstname, pd.player_lastname
""")

In [107]:
manu_defender_sidesline.show(n=100, truncate=False)

+---------+----------------+------------------+------------+
|player_id|player_firstname|player_lastname   |player_count|
+---------+----------------+------------------+------------+
|886      |José Diogo      |Dalot Teixeira    |1           |
|739      |Sergio          |Reguilón Rodríguez|16          |
|18846    |Aaron           |Wan-Bissaka       |2           |
|742      |Raphaël Xavier  |Varane            |3           |
|2467     |Lisandro        |Martínez          |4           |
|18772    |Jonathan Grant  |Evans             |4           |
|891      |Luke Paul Hoare |Shaw              |5           |
|889      |Victor Jörgen   |Nilsson Lindelöf  |2           |
|2935     |Jacob Harry     |Maguire           |3           |
+---------+----------------+------------------+------------+



- 맨체스터 유나이티드 수비수중에 dalot선수 빼고는 20경기 이상 뛴 선수가 없다
- 부상이 많아 못뛴 경기가 많아 수비의 조직력이 느슨해질수밖에 없다
- 중요 선수로 평가받는 shaw,verane,martinez 25경기중에 3번의 부상기록이 올라와져 있다
- 좋은 평가로 받는 평점이 7이 넘는 선수는 3명뿐이 안된다
- 린델로프는 중앙 수비수인데 경합에서 승리하는 승률이 50% 미만이다
- 린델로프는 패스를 많이 하고 정확도와 키패스도 다른 중앙 수비수들보다 많다
- 루크 쇼는 뛸수있는 날에는 선발로 들어가고 윙백이라 드리블시도가 많지만 성공률이 높지는 않다
- 루크쇼는 옐로우카드가 파울범한 횟수에 비해 많다
- 매과이어 선수는 경합승률이 61%이다 달리기가 느린문제때문에 드리블시도는 한번밖에 없다
- 매과이어 선수는 블록태클과 인터셉트 태클이 팀내에서 제일 많다
- 매과이어가 범한 파울이 18개인데 yellow카드가 2개뿐이 없다 위험지역에서는 파울을 잘 안준다
- 매과이어는 어시스트 2개로 공격에 도움을 준다
- 바란은 경합에서 승률이 약 73% 센터백중에서 제일 높은 확률을 보여준다
- 센터백인 바란은 드리블을 잘 하지 않는다
- 패스성공률은 33%로 발밑이좋지는 않다
- Dalot선수는 맨유에서 꾸준히 뛰는 유일한 수비수다
- 윙백인 달롯은 슈팅을 12번을 시도하고 3번의 유효슈팅 1개의 골은 넣고 2번의 어시스트를 했다
- 달롯은 제일 많은 패스횟수인 929번을 하고 키패스를 23번을 넣었다
- 달롯은 태클 인터셉트가 가장많은 27번을 하였고 블록은 10번하였다
- 달롯은 경합승률 58%로 높은 승률을 보여주고 있다.
- 달롯은 드리블 시도를 많이 했고 성공률은 53%로 나쁘지 않은 돌파력을 보여줬다
- 완비사카는 달롯과 같은 윙백이지만 1개의 슈팅만 시도하였다
- 완 비사카는 20개의 인터셉트를 성공하였다
- 완 비사카는 경합 승률 58%로 나쁘지 않는 승률를 가지고있다
- 완 비사카는 드리블 성공률 약 77%이지만 시도를 많이하지않았고 뛴 시간도 885분밖에 되지않는다 25게임중 약 6게임만 뛴게된다
- 마르티네즈 선수는 부상으로 8번의 출전밖에 못하였다
- 마르틴네즈는 패스 정확도 45%로 팀내 중앙 수비수들중  제일 높은 패스 정확도를 보여준다
- 마르티네즈의 경기당 약 6개의 경합중에 경합승률은 47%
- 마르티네즈는 파울을 4번 했는데 옐로우 카드가 3번이다
- 에반스는 경합을 적극적으로 해주면서 50%이상의 승률을 가져오고 있다
- 에반스 드리블은 거의 안하지만 2번했을때 2번다 성공하였다
# 결론
- 좌우 풀백이 3명 밖에 없어 2명이 동시에 부상을 당할시에 문제가 생기기 때문에 윙백을 영입할필요가 있다
- 요즘 축구 트렌드가 패스도 중요하지만 수비수의 목표는 수비를 잘하는게 1순위이다 그런 의미로 린델로프와 마르티네즈선수는 판매를 해야한다
- 마르티네즈 선수는 부상을 자주 당해 경기를 못치르고 패스를 많이하고 잘하지만 수비적인 경합이나 태클부분에서 소극적인 부분이 보여 수비적인 부분과 패스가 되는 다른선수로 대체해야한다.
- 매과이어는 맨체스터 유나이티드의 수비적으로 큰 도움을 주고있다
- 바란도 무난한 수비 스탯을 보여주지만 새로운 센터백을 영입해야한다
- 31살인 매과이어와 30살인 바란 수비수로서 최전성기의 나이지만 어린나이는 아니지만 시즌을 풀로 끌고가기에는 체력적으로나 능력적으로 위험하기때문에 어린 수비수 영입이 필요하다
- 루크 쇼는 공격적인 스탯은 좋지만 수비적인 스탯인 tackles_blocks, tackles_interceptions등이 아쉬움이있다
- 완 비사카 같은 수비력이 있는 윙백 선수를 영입하는것이 필요하다
- 달롯은 많은 경기를 뛰고 공수로 도움을 주고있기때문에 맨체스터 유나이티드는 지킬 필요가 있다
- 맨체스터 유나이티드 현재 전술때문에 패스성공률이 낮다

In [108]:
manu_player_midfielder_stat = spark.sql("""SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname,pf.player_birth,pf.player_nationality,TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, ps.*, AVG(ps.game_rating) OVER () as avg_game_rating,duels_won/duels_total,dribbles_success/dribbles_attempts,game_lineups/game_appearances,passes_total/(game_minutes/90),passes_key/(game_minutes/90),tackles_blocks/(game_minutes/90),tackles_interceptions/(game_minutes/90),duels_total/(game_minutes/90),dribbles_attempts/(game_minutes/90),game_minutes/90 total_games,total_shots/(game_minutes/90),shots_on/(game_minutes/90),goals_assists/(game_minutes/90),passes_key/passes_total,shots_on/total_shots,goals_assists/passes_key,fouls_drawn/(game_minutes/90),fouls_committed/(game_minutes/90),cards_yellow/(game_minutes/90),cards_yellow/fouls_committed,cards_yellowred/(game_minutes/90),cards_red/fouls_committed,total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
FROM england_player_stat ps
JOIN team_df tf ON ps.team_id = tf.team_id
JOIN player_df pf ON ps.player_id = pf.player_id
WHERE tf.team_name LIKE 'Manchester United' AND ps.game_position LIKE 'Midfielder' AND ps.game_appearances > 1;

""")

In [109]:
manu_player_midfielder_stat.show()

24/03/12 16:04:09 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 16:04:09 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 16:04:09 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 16:04:09 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 16:04:09 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 16:04:09 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 1

+-----------------+----------------+-----------------+------------+------------------+----------+---------+-------+---------+-------------+----------------+------------+------------+-----------+-------------+-----------+------------+------+-------+---------+-----------+--------+-----------+--------------+-------------+-----------+------------+----------+---------------+-------------+--------------+---------------------+-----------+---------+-----------------+----------------+-------------+-----------+---------------+------------+---------------+---------+-----------+----------------+--------------+--------------+-------------+---------------+-------------------------+--------------------------------------+---------------------------------+------------------------------------+----------------------------------+--------------------------------------+---------------------------------------------+-----------------------------------+-----------------------------------------+----------------

In [110]:
manu_midfielder_sidesline = spark.sql("""SELECT ps.player_id, pd.player_firstname, pd.player_lastname,COUNT(ps.player_id) as player_count
FROM england_player_stat p
JOIN player_df pd ON p.player_id = pd.player_id
JOIN player_sideline ps ON p.player_id = ps.player_id
WHERE p.team_id = 33 AND p.game_position LIKE 'Midfielder' AND p.game_appearances > 1 AND ps.sideline_start > '2023-08-10'
GROUP BY ps.player_id, pd.player_firstname, pd.player_lastname



""")

In [111]:
manu_midfielder_sidesline.show()

+---------+----------------+-----------------+------------+
|player_id|player_firstname|  player_lastname|player_count|
+---------+----------------+-----------------+------------+
|     1485|    Bruno Miguel| Borges Fernandes|           1|
|    19220|      Mason Tony|            Mount|           8|
|      747| Carlos Henrique|         Casimiro|           2|
|       74|          Sofyan|          Amrabat|          12|
|      174|       Christian|Dannemann Eriksen|           3|
|      903|   Scott Francis|        McTominay|           1|
|    70078|         Facundo|Pellistri Rebollo|           1|
+---------+----------------+-----------------+------------+



- 마이누는 어린 18살 선수인데 많은 기회를 받고 있다
- 마이누는 경기당 약 한개 이상의 인터센트를 성공하고있다
- 마이누는 경기당 10개의 경합중 50%의 경합승률을 보여주고있다
- 페르난데스 선수는 총 25경기중에 24경기를 출장하며 높은 출장률을 보여주고 있다
- 페르난데스의 공격적으로 뛰는 움직임도 있지만 key패스 생성이 제일 높다
- 페르난데스는 슛팅은 경기당 2개 가까이 쏘지만 골문쪽으로 가는 비율이 높은 편은 아니다
- 페르난데스는 2경기당 1개의 비율로 어시스트를 하였다
- 페르난데스는 경기당 3개의 키패스를 넣어주었다
- 페르난데스는 드리블 성공률이 50% 높지 않다
- 페르난데스는 패스 성공률 2위를 기록하고있다
- 펠리스트리는 출전기록은 9번이지만 대부분 교체로 출전하였다
- 에릭센은 만 32살이다
- 에릭센은 로테이션 자원으로 선발과 교체를 오가고 있다
- 에릭세는 1경기를 90분으로 쳤을때 경기당 1번에 가까운 슈팅을 했고 80%가 유효슈팅이되었다
- 에릭센의 패스는 1경기 90분 기준 55개의 패스 팀내 미드필더 4위를 기록하고 있다
- 에릭센의 키패스는 1경기 90분 기준 1.48로 경기당 평균 한번의 키패스를 주고있다 미드필더 팀내기준 2위를 기록하고있다
- 하지만 에릭센의 패스 성공률은 29%에 그쳤다
- 에릭센의 인터셉트는 다른 팀내 미드필더 선수보다 월든히 낮을 0.3을 기록하고있다
- 에릭센은 경합 90분당 6번으로 소극적인 모습을 보이고있다
- 에릭센의 경합 성공률도 32%에 그쳤다
- 에릭센은 드리블을 많이 시도하지는 않지만 성공률은 75%로 괜찮은 성공률을 보여주고있다
- 맥토미니는 27살로 전성기인 나이다
- 맥토미니는 선발과 교체를 오고가는 선수이다
- 맥토미니는 총 22경기에 출전했지만 90분기준 약 14경기정도 뛰었다
- 맥토미니는 출전 시간중에 27번의 슈팅을 시도하였고 17번의 유효슈팅을 기록하였다 이는 전체슈팅의 60%가 유효슈팅으로 기록되었다
- 맥토미니는 슈팅을 90분기준 경기당 평균 2개를 시도하였다
- 맥토미니는 약 14경기정도 뛰었는데 그 안에 7골을 득점하고 1개의 도움을 기록하였다
- 맥토미니는 패스는 경기당 29개로 팀내 미드필더 뒤에서 2위를 기록하고 패스 정확도는 16%를 기록하고있다
- 맥토미니는 드리블시도가 13번있었고 7번을 성공시켜 성공률 약 54%를 기록하였다
- 맥토미니는 90분 기준 평균 약 1개의 드리블 시도있다
- 맥토미니는 블록 태클이 7개로 경기당 평균 0.5개를 기록하고있고 팀내 미드필더 6위를 기록하고있다
- 맥토미니는 인터셉트가 10개로 경기당 평균 0.72개를 기록하였고 팀내 미드필더 5위를 기록하고있다
- 암라바트는 27살의 전성기에 있는 선수이다
- 암라바트는 12경기를 나왔지만 9경기 선발에 3경기는 교체로 출전하엿고 90분기준 7경기정도의 시간을 뛰었다
- 암라바트는 90분기준 경기당 평균 62개의 패스를 했다 패스 정학도는 37%를 기록하고있다
- 암라바트는 90분 기준 경기당 평균 약 10개의 경합을하고 승률은 53%를 기록하고 있다
- 암라바트는 드리블을 5개중 4번을 성공시키며 80%의 성공률을 보여주고있다
- 암라바트는 블록을 90분 기준 경기당 평균 0.4개를 기록하고 있다 
- 암라바트는 인터셉트를 90분 기준 경기당 평균 0.73개를 기록하고 있다
- 암라바트는 패스중 키패스 비율이 0.9%를 기록하고있다
- 암라바트는 경기당 평균 0.29개의 슈팅 기록하고있다 
- 카세미루는 32살로 전성기에서 꺽이는 나이다
- 카세미루가 뛴 12경중에서는 12경기 모두 선발로뛴 선발자원이다
- 카세미루는 17번의 슈팅 시도 5개의 유효슈팅, 유효슈팅중 1골을 기록하였다
- 카세미루의 유효슈팅 성공률은 29%이다
- 카세미루는 90분기준 경기당 평균 59개의 패스로 팀내 미드필더 2위를 기록하고있고 패스정확도는 43%로 팀내 미드필더중에서 제일 높다
- 카세미루는 90분기준 경기당 평균 0.85개의 블록과 인터셉트를 기록하였고 팀내 미드필더 1위와 2위를 기록하고있다
- 카세미루는 키패스 9개로 90분기준 평균 0.85개를 기록중이고 어시스트는 1개기록하고있다
- 카세미루의 총패스중 키패스는 1.4%로 팀내 미드필더 뒤에서 3번째를 기록중이다
- 카세미루는 시즌중 2번의 부상이력이 있다
- 마운트는 만 25세로 젊은 선수이다
- 마운트는 시간으로 5경기 미만으로 뛴 선수이다
- 마운트는 8경기중에 4경기 선발 4경기 교체로 출전하였다
- 마운트는 슛팅 한번에 유효슈팅은 없었다
- 마운트는 90분기준 경기당 평균 패스 33개로 팀내 미드필더 뒤에서 3위를 기록하고있다
- 마운트는 패스정확도 15%를 기록하고있다
- 마운트는 블록은 경기당 평균 0.22로 팀내 미드필더 뒤에서 2위를 기록하고있다
- 마운트는 총패스중 3%가 키패스로 기록되었다
- 마운트는 90분경기 기준 경기당 평균 1번이상의 키패스를 해주었다
- 마운트는 경합승률 49%를 기록하였다
- 마운트는 드리블을 6번 시도하엿고 4번을 성공하면 66%를 기록하였다
- 마운트는 6번의 파울을 얻어내며 경기당 평균 한번이상의 파울을 얻어낸다
- 한니발은 21살의 어린선수이다
# 결론
- 마이누는 아직 좋은 활약이라고 할 수는 없지만 아직 18살로 성장을 더 할수있기때문에 꾸준히 출전 기회를 주며 키워볼만하다
- 미드필더는 주어진 역할에 따라 다르겠지만 공격적인것과 수비적인것을 둘다 가져가야하는 포지션이기때문에 수비위치를 좀 더 잘 잡아서 block횟수를 좀 늘리면 수비적인 역할에서 좀 더 성장할수있다
- 페르난데스 선수는 맨체스터 유나이티의 공격의 핵심이다 공격의 중요한 요소들은 페르난데스 선수가 상위권을 차지하고있다
- 페르난데스는 맨체스터에서 중요한 선수기때문에 파울을 해 카드받는것을 조심해야한다
- 페르난데스는 수비적인 통계가 좋지않지만 수비의 부담을 주는것보단 공격으로 더 뽑낼수있도록 더 공격에 집중하는것이 좋을것이다 하지만 슈팅의 유효슈팅 확률을 좀 더 높일 필요가 있다
- 펠리스트리는 현재 임대를 가있는 상태이다
- 에릭센은 패스로 팀에 도움을 줄수있는선수다 하지만 수비적으로 경합이나 인터셉트가 너무 낮아 다른 선수에게 부담을 준다
- 에릭센은 나이도 감안하면 주전으로 쓰기보다는 공격적인 선수가 필요할때 교체자원으로 쓰면 좋다
- 만약 적절한 금액에 판매가 가능하다면 판매도 고려해볼수있다
- 맥토미니는 공격적인 역할에 특화되어있고 수비적인 능력은 팀내에서 중간정도 하고있고 패스능력은 낮다
- 맥토미니는 교체 멤버로서 좋은 카드라고 판단되지만 주전으로 삼기에는 더 성장해야할 부분이 많다
- 암라바트는 위치는 수비형 미드필드로 뛰고 있어 경기당 평균 패스 갯수가 제일 많다
- 암라바트는 패스정확도를 높일 필요가 있다
- 암라바트는 수비형 미드필더로써 인터셉트와 블록을 늘리면서 수비적으로 더 도움을 줄 필요가 있다
- 암라바트는 수비적으로나 공격적으로 도움이 크게 되지 않는다
- 암라바트는 이번시즌이 첫시즌이므로 더 지켜봐야한다
- 카세미루 위치상 안전한 패스를 많이해 패스 성공률이 높지만 키패스는 전체패스중 1%만 차지 안한다
- 카세미루는 수비적으로 미드필더로써 연결고리역할은 괜찮게 해주고 있지만 나이를 고려하면 카세미루만 믿고 가기에는 불안정하다
- 마운트는 뛴 경기가 부족해 더 지켜봐야한다
- 출전한 경기에서는 키패스를 잘 주는건 좋지만 더 공격적으로 슛팅을 더 시도하고 패스도 더 정확하게 줄 필요가 있다.
- 마운트는 이번시즌 첫 시즌이라 더 지켜봐야한다
- 중원 싸움중 경합에서 확실히 공을 뺐어올만한 선수가 부족하다
- 마운트와 페르난데스를 동시에 사용할시 수비력이 좋은 선수를 써야한다
- 한니발은 어린선수로 더 지켜봐야한다
- 수비적으로 더 뛰어난 수비형미드필더 영입이 필요하다

In [112]:
manu_player_attacker_stat = spark.sql("""SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname,pf.player_birth,pf.player_nationality,TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, ps.*, AVG(ps.game_rating) OVER () as avg_game_rating,duels_won/duels_total,dribbles_success/dribbles_attempts,game_lineups/game_appearances,passes_total/(game_minutes/90),passes_key/(game_minutes/90),tackles_blocks/(game_minutes/90),tackles_interceptions/(game_minutes/90),duels_total/(game_minutes/90),dribbles_attempts/(game_minutes/90),game_minutes/90 total_games,total_shots/(game_minutes/90),shots_on/(game_minutes/90),goals_assists/(game_minutes/90),passes_key/passes_total,shots_on/total_shots,goals_assists/passes_key,fouls_drawn/(game_minutes/90),fouls_committed/(game_minutes/90),cards_yellow/(game_minutes/90),cards_yellow/fouls_committed,cards_yellowred/(game_minutes/90),cards_red/fouls_committed,total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
FROM england_player_stat ps
JOIN team_df tf ON ps.team_id = tf.team_id
JOIN player_df pf ON ps.player_id = pf.player_id
WHERE tf.team_name LIKE 'Manchester United' AND ps.game_position LIKE 'Attacker' AND ps.game_appearances > 1;

""")

In [113]:
manu_player_attacker_stat.show()

24/03/12 16:04:10 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 16:04:10 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 16:04:11 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 16:04:11 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 16:04:11 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 16:04:11 WARN WindowExec: No Partition Defined for Window operation! Moving all data to a single partition, this can cause serious performance degradation.
24/03/12 1

+-----------------+----------------+-----------------+------------+------------------+----------+---------+-------+---------+-------------+----------------+------------+------------+-----------+-------------+-----------+------------+------+-------+---------+-----------+--------+-----------+--------------+-------------+-----------+------------+----------+---------------+-------------+--------------+---------------------+-----------+---------+-----------------+----------------+-------------+-----------+---------------+------------+---------------+---------+-----------+----------------+--------------+--------------+-------------+---------------+-------------------------+--------------------------------------+---------------------------------+------------------------------------+----------------------------------+--------------------------------------+---------------------------------------------+-----------------------------------+-----------------------------------------+----------------

In [114]:
manu_attacker_sidesline = spark.sql("""SELECT ps.player_id, pd.player_firstname, pd.player_lastname,COUNT(ps.player_id) as player_count
FROM england_player_stat p
JOIN player_df pd ON p.player_id = pd.player_id
JOIN player_sideline ps ON p.player_id = ps.player_id
WHERE p.team_id = 33 AND p.game_position LIKE 'Attacker' AND p.game_appearances > 1 AND ps.sideline_start > '2023-08-10'
GROUP BY ps.player_id, pd.player_firstname, pd.player_lastname



""")

In [115]:
manu_attacker_sidesline.show()

+---------+----------------+---------------+------------+
|player_id|player_firstname|player_lastname|player_count|
+---------+----------------+---------------+------------+
|      909|          Marcus|       Rashford|           3|
|      908|  Anthony Jordan|        Martial|           2|
|     9971|  Antony Matheus|     dos Santos|           1|
|       18|     Jadon Malik|         Sancho|           8|
|   288006|          Rasmus|Winther Højlund|           4|
+---------+----------------+---------------+------------+



- 래쉬포드는 26살은 젊은 공격수이다
- 래쉬포드는 25경기 기준 24번을 출전하였고 시간상으로는 19경기를 뛰었다
- 래쉬포드는 총 33개의 슈팅중에 17개의 유효슈팅을 기록하고 4골의 필드골 1골의 패널티킥을 넣고 2개의 어시스트를 기록하였다   
- 래쉬포드는 90분기준 경기당 평균 1.7개의 슈팅을 시도하였다
- 래쉬포드는 경기당 평균 유효슈팅 0.87개를 기록하고있다
- 래쉬포드는 유효슈팅이 득점이 된 골은 29%정도 된다
- 래쉬포드는 516개의 패스중에 20개의 키패스를 하고 16%의 정확도를 갖고있다
- 래쉬포드는 90분기준 경기당 평균 26개의 패스를 한다
- 래쉬포드는 90분기준 경기당 평균 1개의 키패스를 준다
- 래쉬포드의 전체패스중 키패스비율은 3%로 팀내 공격수중 2번째로 낮다
- 래쉬포드는 202개의 경합중 84개를 이기면서 경합승률 41%를 기록하고잇다
- 래쉬포드는 90분경기기준 경기당 평균 10개의 경합을 시도한다
- 래쉬포드는 드리블을 84번 시도하였고 41개의 드리블을 성공시키며 성공률 49%를 기록하고있다
- 래쉬포드는 90분기준 경기당 평균 4번의 드리블 시도를 한다
- 래쉬포드가 파울을 얻어낸 수는 경기당 평균 0.9개로 팀내 공격수에서 뒤에서 1등을 기록하고있다
- 래쉬포드는 90분기준 경기당 평균 0.05개의 블록을 했다 이는 팀내 공격수 3위이다
- 래쉬포드는 90분기준 경기당 평균 0.15개의 인터셉트을 했다 이는 팀내 공격수 뒤에서 3위이다
- 래쉬포드가 파울을 한 수는 경기당 평균 0.9개로 팀내 공격수에서 뒤에서 1등을 기록하고있다
- 회이룬은 21살의 어린 공격수이다
- 회이룬은 25경기 기준 20경기를 뛰었고 시간상으로 16경기를 뛰었다
- 회이룬은 25개의 슈팅을 하였고 16개의 유효슈팅을 만들고 그중 7개를 골로 만들었고 2개의 어시스트를 했다
- 회이룬은 팀내 득점순위 1위 이고 공격포인트생성 1위이다
- 회이룬은 90분경기 기준 경기당 평균 1.5개의 슈팅을 한다
- 회이룬은 90분경기 기준 경기당 평균 0.9개의 유효슈팅을 한다 이는 팀내 공격수 1위 이다
- 회이룬의 전체슈팅중 64%가 유효슈팅으로 기록되었다
- 회이룬의 유효슈팅중 43% 득점으로 이어졌다
- 회이룬은 256개의 패스중에 17개의 키패스를 하고 10%의 정확도를 기록하고있다
- 회이룬은 90분 경기 기준 경기당 평균 16개의 패스를한다
- 회이룬은 90분 경기 기준 경기당 평균 1개의 키패스를한다
- 회이룬의 키패스는 전체 패스의 6%로 팀내 공격수 2위를 기록하고있다
- 회이룬은 경합을 149회 시도하고 60개를 이겼다 이는 승률 40%이다
- 회이룬은 경합을 90분기준 경기당 평균 9개의 경합을 시도한다
- 회이룬은 드리블을 22번 시도하여 12번을 성공하였다 이는 57%의 성공률로 팀내 공격수 1위이다
- 회이룬은 파울을 22번 얻어내면서 90분기준 경기당 1번이상의 파울을 얻어냈다
- 회이룬은 10번의 파울을범하였고 옐로우 카드를 2번 얻었다
- 안토니는 24살의 전성기로 가고있는 선수이다
- 안토니는 19경기를 출장하고 11번을 선발선수로 뛰고 10경기정도의 시간을 뛰었다
- 안토니는 13번의 슈팅을 시도하고 6개의 유효슈팅을 성공하고 0골을 득점하였다
- 안토니는 90분경기 기준 경기당 평균 한개의 슈팅을 가져갔고 0.58개의 유효슈팅을 때렸다 이는 팀내 공격수 뒤에서 2번째이다
- 안토니의 슈팅이 유효슈팅이 되는 비율은 전체 슈팅의 46%로 팀내 공격수 뒤에서 2번째를 기록하고있다
- 안토니는 397개의 패스를 하고 16개의 키패를 하고 어시스트는 0개이다
- 안토니의 패스 정확도는 17%이다
- 안토니는 90분경기 기준 경기당 평균 38개의 패스를 했다
- 안토니 90분경기 기준 경기당 평균 1개이상의의 키패스를 했다
- 안토니의 전체패스중 키패스의 비율은 4% 팀내 공격수 뒤에서 3번째를 차지하고있다
- 안토니의 태클은 1개의 블록과 8개의 인터센트를 기록하고있다
- 안토니는 90분경기 기준 경기당 평균 0.09개의 블록과 0.77개의 인터셉트를 했다
- 안토니는 경합을 146번을 시도하였고 68번 경합에서 이겼다 이는 승률 47% 팀내 공격수 1위이다
- 안토니는 90분경기 기준 경기당 평균 14경합을 시도한다 이는 팀내 공격수 2위의 기록이다
- 안토니는 드리블을 42번 시도하였고 17번 성공하였다 이는 성공률 40%이다
- 안토니는 90분경기 기준 경기당 평균 4번의 드리블를 시도한다 이는 팀내 공격수 2위의 기록이다
- 안토니는 23개의 파울을 얻어냈다 이는 90분기준 경기당 평균2개의 파울을 얻어낸 기록이다
- 안토니는 11개의 파울을 했다 이는 90분기준 경기당 평균1개의 파울을 한다
- 안토니의 파울의 비율중 옐로우카드받는 비율이 27% 팀내 공격수중 제일 높다
- 가르나초 19살의 어린선수이다
- 가르나초는 25경기중에서 23경기를 출전하고 17번 선발을 할정도로 중용받고있다
- 가르나초는 시간상으로는 약 17경기정도의 시간을 뛰었다
- 가르나초는 37개의 슈팅을 했고 15개의 유효슈팅을 성공하고 5골을 득점하였다 
- 가르나초는 90분 기준 경기당 평균 2번의 슈팅을 시도하였다
- 가르나초는 90분 기준 경기당 평균 0.88번의 유효슈팅을 하였다 이는 팀내 공격수 2위에 해당되는 수치이다
- 가르나초는 유효슈팅중 골들어간 수치는 33%를 기록하고있다
- 가르나초는 유효슈팅률은 약 41%정도 기록하고있다
- 가르나초는 489번의 패스를 하고 32개의 키패스를 기록하고 패스정확도는 16%를 기록하고있다
- 가르나초는 90분 기준 경기당 평균 28번의 패스를 하고 1개의 키패스를 준다
- 가르나초는 90분 기준 경기당 평균 0.17번의 어시스트를 했다 이는 팀내 공격수 1위의 수치이다
- 가르나초의 전체패스의 6%가 키패스로 팀내 공격수 1위의 수치이다
- 가르나초는 1개의 블록을하고 이는 90분 기준 경기당 평균 0.05번의 블록를 했고 팀내 공격수 2위의 수치이다
- 가르나초는 10개의 인터셉트를 하고 이는 90분 기준 경기당 평균 0.59번의 블록를 했고 팀내 공격수 3위의 수치이다
- 가르나초는 경합을 218번 도전하였고 88번 승리하였다 이는 경합승률 40%를 나타낸다
- 가르나초는 90분기준 경기당 평균 약 13번정도 경합하였다 이는 팀내 공격수 3위를 차지하는 기록이다
- 가르나초는 76번의 드리블을 시도하였고 32번의 드릐블을 성공하였다 이는 42%에해당한다
- 가르나초는 90분경기 기준 경기당 평균 4.5개의 드리블을 시도하였다
- 가르나초가 파울을 얻어낸 갯수는 36번으로 팀내 공격수 1위를 기록하였다
- 가르나초는 16번의 파울을 하고 4번의 경고를 받았다
- 가르나초는 90분기준 경기당 평균 2번의 파울을 얻어내고 1번의 파울을 했다
- 가르나초의 파울 25%를 경고를 받았다
- 산초는 3번을 출전하였지만 선발을 한적은 없고 다 교체로 출전하여 1경기미만의 시간을 뛰었다
- 산초는 슈팅을 시도한 기록이 없다
- 산초는 41번의 패스를 하였고 1개의 키패스를 기록하였고 11%의 정확도를 기록하였다
- 산초는 90분 경기 기준 경기당 평균 48번의 패스를 한 기록이지만 90분미만으로뛰어서 쌓인데이터가 부족하다 기록상으론 팀내 공격수 1위의 갯수이다
- 산초는 90분 경기 기준 경기당 평균 1번의 키패스를 한 기록이다
- 산초의 키패스 생성률은 2%로 팀내 공격수 뒤에서 1위를 기록하고있다
- 산초는 1번의 인터셉트를 기록하였다
- 산초는 16번의 경합중 6번을 승리하였다 이는 37.5% 승률을 기록하고있고 팀내 공격수 뒤에서 1등을 기록하고있다
- 산초는 90분 경기 기준 경기당 평균 18번의 경합을 한 기록이다
- 산초는 6번의 드리블을 시도하였고 3번을 성공하며 성공률 50%를 기록하였다
- 산초는 90분 경기 기준 경기당 평균 7번의 드리블시도를 한 기록이다
- 산초는 2개의 파울를 얻어냈다 
- 마샬은 13경기를 출전하였지만 5경기만 선발로 뛴 로테이션 자원이고 약 5경기에 해당되는 시간을 뛰었다
- 마샬은 5개의 슈팅을 때렸고 4개의 유효슈팅 1개의 골을 기록하였다
- 마샬의 전체 슈팅의 80%가 유효슈팅으로 기록되었다
- 마샬은 90분기준 경기당 평균 1번의 슈팅을 시도했다
- 마샬은 90분기준 경기당 평균 0.8번의 유효슈팅을 기록하였다
- 마샬은 총패스 137개를 하고 6번의 키패스 9%의 정확도로 정확도 팀내 공격수 뒤에서 1등을 기록하였다
- 마샬은 90분경기기준 경기당평균 약 28번의 패스를 하였다
- 마샬은 90분경기기준 경기당평균 약 1번의 키패스를 하였다
- 마샬의 전체패스 갯수중 키패스 비율은 4% 팀내 3위를 기록하고있다
- 마샬은 인터셉트와 블록은 기록하지 못했다
- 마샬은 경합 53번중 20번을 승리하였다 이는 승률 37%로 팀내 공격수 뒤에서 2위를 기록하고있다
- 마샬은 90분기준 경기당 평균 약 11번의 경합을 시도하였다
- 마샬은 4번 시도하였고 2번을 성공하였다 성공률 50%를 기록하였다
- 마샬은 90분경기 기준 경기당 평균 0.8번를 시도하였다 
- 마샬은 6번의 파울을 얻어내었다
- 마샬은 90분경기 기준 경기당 평균 1번의 파울을 얻어내었다
- 마샬은 3번의 파울을 하엿다
- 마샬은 90분경기 기준 경기당 평균 0.6번의 파울을 얻어내었다

# 결론
- 래쉬포드는 현재 주전 선수로 중용받고있지만 활약은 아쉽다
- 래쉬포드의 골 결정력이 29%로 낮은 상태이다
- 래쉬포드는 경합과 수비도 소극적으로하고있다
- 래쉬포드는 드리블을 많이치지만 성공률을 높일필요가 있다
- 래쉬포드는 현재 활약이 아쉽지만 지난 시즌에 했던것도 있고 좀 지켜봐야할 필요가 있다
- 회이룬은 최전방 공격수로서 공격적 포인트를 어느정도 해주고있다
- 드리블 성공율,파울얻어내기,유효슈팅 많은 공격적 부분이 팀내 공격수 상위권이 지키고이싸
- 경합승률을 늘리고 버티고 정확한 패스를 준다면 좀더 좋은 선수가 될것이다
- 회이룬은 어린선수 유망주로서 첫시즌인 만큼 어느정도 괜찮게 활약하고있다
- 안토니는 적극적인 선수이다 경합이며 패스시도며 드리블시도면 높은 수치를 갖고있다
- 하지만 안토니는 드리블성공률이높지않으며 패스 성공률도 낮다 또한 공격수로서 제일 중요한 공격포인트가 0개이다
- 2년차로서 안토니는 판매를 하는것이 좋을거같다
- 가르나초는 어린나이에 맨체스터 유나이티드에서 중용을 받고있으면서 좋은 활약을 하고있다
- 가르나초는 팀내에서 공격포인트 2위를 기록하고있고 어시스트는 3개를하면서 팀내 2위를 기록하고있다 또한 활발하게 움직이면서 파울을 많이 얻어내고있다
- 가르나초는 팀내 선수중에 질좋은 패스를 주고있다
- 산초는 74분을 뛰었지만 뛰는 중에 슈팅을 기록을 못한정도로 안좋은 활약을 했었다
- 산초의 74분 기록중 공격 생성력도 안좋아 키패스 갯수도 적고 슈팅을 못하였다
- 산초의 74분 중에 경합 성공률도 가장낮고 태클 한번에 인터셉트 한번만 할 정도로 수비적으로도 도움이 크게되지않는다
- 산초는 팀내 공격서들중 제일많이 시도한 기록을 갖고있다
- 산초는 판매를 하는게 좋다다
- 마샬은 유효슈팅 생성률은 좋지만 전체적으로 게으르다
- 마샬은 경합승률,인터셉트,블록,드리블시도횟수, 파울얻어냄등을 팀내 공격수안에서 하위권을 기록하였다
- 마샬을 판매를 할 필요가 있다
- 윙포워드와 최전방 공격수의 영입이 필요하다

# 전체결론
- 20경기 이상뛴 선수가 몇명 없을 만큼 부상과 경기력에 큰영향이 있다
- 수비는 윙백포지션에는 수비형에 가까운 선수를 영입하면서 상대에 따라서 공격적으로 할때와 수비적으로 할때같은 다양한 전술을 사용할수있으므로 윙백선수를 영입하는게 좋다
- 중앙수비는 린델로프의 활약도가 약하므로 판매를하고 새로운 수비수 한명을 영입하는게 좋을것이다 30대이상인 선수를 주전으로 쓰기에는 오랫동안 사용을 못하니 미래를 위해서라도 중앙수비수 영입이 필요하다
- 수비형 미드필더 한명을 영입하므로 경합에서 공을 따고 상대방 공격에서 블록이든 인터셉트든 수비쪽에서 더 도움을 줄수잇는 선수가 있으면 페르난데스와 마운트를 지원할수있을것이다
- 공격수들중에 경기당 슈팅갯수를 2개 이상 가져간 공격수는 가르나초 한명밖에 없다 이는 골넣은 기회도 그렇게 많지 않았다는 뜻이다.
- 윙어 공격수를 영입할 필요가 있다
- 맨체스터 유나이티드 공격진에는 공격포인트가 10개가 넘는 선수가 한명도 없다
- 안토니와 산초 마샬을 판매를 하고 드리블로 사이드를 뚫을수있는 윙어와 회이룬을 도움을 줄수있는 최전방 공격수 한명씩 영입해야한다
- 감독 전술때문인지 맨체스터 유나이티드의 패스 정확도가 전체적으로 낮다 50%넘는선수는 없다.


In [116]:
world_player_defender_suggest = spark.sql("""With players as(SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname,pf.player_birth,pf.player_nationality,TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, eps.*,duels_won/duels_total,dribbles_success/dribbles_attempts,game_lineups/game_appearances,passes_total/(game_minutes/90),passes_key/(game_minutes/90),tackles_blocks/(game_minutes/90),tackles_interceptions/(game_minutes/90),duels_total/(game_minutes/90),dribbles_attempts/(game_minutes/90),game_minutes/90 total_games,total_shots/(game_minutes/90),shots_on/(game_minutes/90),goals_assists/(game_minutes/90),passes_key/passes_total,shots_on/total_shots,goals_assists/passes_key,fouls_drawn/(game_minutes/90),fouls_committed/(game_minutes/90),cards_yellow/(game_minutes/90),cards_yellow/fouls_committed,cards_yellowred/(game_minutes/90),cards_red/fouls_committed,total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
FROM england_player_stat eps
JOIN player_df pf ON eps.player_id = pf.player_id
JOIN team_df tf ON eps.team_id = tf.team_id
Union 
SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname,pf.player_birth,pf.player_nationality,TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, sps.*,duels_won/duels_total,dribbles_success/dribbles_attempts,game_lineups/game_appearances,passes_total/(game_minutes/90),passes_key/(game_minutes/90),tackles_blocks/(game_minutes/90),tackles_interceptions/(game_minutes/90),duels_total/(game_minutes/90),dribbles_attempts/(game_minutes/90),game_minutes/90 total_games,total_shots/(game_minutes/90),shots_on/(game_minutes/90),goals_assists/(game_minutes/90),passes_key/passes_total,shots_on/total_shots,goals_assists/passes_key,fouls_drawn/(game_minutes/90),fouls_committed/(game_minutes/90),cards_yellow/(game_minutes/90),cards_yellow/fouls_committed,cards_yellowred/(game_minutes/90),cards_red/fouls_committed,total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
from spain_player_stat sps
JOIN player_df pf ON sps.player_id = pf.player_id
JOIN team_df tf ON sps.team_id = tf.team_id

Union 
SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname,pf.player_birth,pf.player_nationality,TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, ips.*,duels_won/duels_total,dribbles_success/dribbles_attempts,game_lineups/game_appearances,passes_total/(game_minutes/90),passes_key/(game_minutes/90),tackles_blocks/(game_minutes/90),tackles_interceptions/(game_minutes/90),duels_total/(game_minutes/90),dribbles_attempts/(game_minutes/90),game_minutes/90 total_games,total_shots/(game_minutes/90),shots_on/(game_minutes/90),goals_assists/(game_minutes/90),passes_key/passes_total,shots_on/total_shots,goals_assists/passes_key,fouls_drawn/(game_minutes/90),fouls_committed/(game_minutes/90),cards_yellow/(game_minutes/90),cards_yellow/fouls_committed,cards_yellowred/(game_minutes/90),cards_red/fouls_committed,total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
from italy_player_stat ips
JOIN player_df pf ON ips.player_id = pf.player_id
JOIN team_df tf ON ips.team_id = tf.team_id

Union 
SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname,pf.player_birth,pf.player_nationality,TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, gps.*,duels_won/duels_total,dribbles_success/dribbles_attempts,game_lineups/game_appearances,passes_total/(game_minutes/90),passes_key/(game_minutes/90),tackles_blocks/(game_minutes/90),tackles_interceptions/(game_minutes/90),duels_total/(game_minutes/90),dribbles_attempts/(game_minutes/90),game_minutes/90 total_games,total_shots/(game_minutes/90),shots_on/(game_minutes/90),goals_assists/(game_minutes/90),passes_key/passes_total,shots_on/total_shots,goals_assists/passes_key,fouls_drawn/(game_minutes/90),fouls_committed/(game_minutes/90),cards_yellow/(game_minutes/90),cards_yellow/fouls_committed,cards_yellowred/(game_minutes/90),cards_red/fouls_committed,total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
from germany_player_stat gps
JOIN player_df pf ON gps.player_id = pf.player_id
JOIN team_df tf ON gps.team_id = tf.team_id

Union 
SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname, pf.player_birth, pf.player_nationality, TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, fps.*, duels_won/duels_total, dribbles_success/dribbles_attempts, game_lineups/game_appearances, passes_total/(game_minutes/90), passes_key/(game_minutes/90), tackles_blocks/(game_minutes/90), tackles_interceptions/(game_minutes/90), duels_total/(game_minutes/90), dribbles_attempts/(game_minutes/90), game_minutes/90 total_games, total_shots/(game_minutes/90), shots_on/(game_minutes/90), goals_assists/(game_minutes/90), passes_key/passes_total, shots_on/total_shots, goals_assists/passes_key, fouls_drawn/(game_minutes/90), fouls_committed/(game_minutes/90), cards_yellow/(game_minutes/90), cards_yellow/fouls_committed, cards_yellowred/(game_minutes/90), cards_red/fouls_committed, total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
FROM france_player_stat fps
JOIN player_df pf ON fps.player_id = pf.player_id
JOIN team_df tf ON fps.team_id = tf.team_id)
                                          
select * from players
WHERE game_position LIKE 'Defender' AND game_minutes/90 > 12 AND tackles_blocks/(game_minutes/90)>0.55 AND tackles_interceptions/(game_minutes/90) >0.55 AND dribbles_attempts/(game_minutes/90)> 1;


""")

In [117]:
world_player_defender_suggest.show()

24/03/12 16:04:17 WARN DAGScheduler: Broadcasting large task binary with size 1187.1 KiB


+-----------------+--------------------+--------------------+------------+------------------+----------+---------+-------+---------+-------------+----------------+------------+------------+-----------+-------------+-----------+------------+------+-------+---------+-----------+--------+-----------+--------------+-------------+-----------+------------+----------+---------------+-------------+--------------+---------------------+-----------+---------+-----------------+----------------+-------------+-----------+---------------+------------+---------------+---------+-----------+----------------+--------------+--------------+-------------+-------------------------+--------------------------------------+---------------------------------+------------------------------------+----------------------------------+--------------------------------------+---------------------------------------------+-----------------------------------+-----------------------------------------+------------------+------

- 센터백으로 수비수 반 헤케를 추천한다 23살이 어린나이와 브라이튼에서 꾸준히 출전하는 선수이다
- 반 헤케는 90분기준 매경기 평균 8번정도의 경합을 도전하는데 경합 승률 67%를 기록중인 선수이고 블록과 인터센트를 90분기준 매경기 한개 이상씩해주는 선수이다
- 반헤케는 수비수인데 파울을 얻어내는것도 13번으로 많다
- 반 헤케는 수비수인데 90분기준 매경기 평기 97번의 패스를 할정도로 팀안에서 패스를 담당하고있다
- 이미 잉글랜드에서 뛰고 있어서 적응시간이 덜 필요하다
- 또 다른 센터백 수비수로는 자가두 선수를 추천한다 24살의 선수와 팀안에서 19번을 출전하였다
- 자가두 선수는 90분경기 기준 매경기 평균 경합을 7번 이상 도전하는데 하는데 경합승률이 70%일 정도로 높은 경합승률을 가지고 있다
- 자가두는 90분 기준 0.5개의 슈팅을 시도한다 유효슈팅은로 변환은 아쉽다
- 파울을 많이 범하긴한다 하지만 경고카드한 1개일 정도로 위험한 파울은 하지않는다
- 90분 경기 기준 매경기 평균 1개이상의 블로과 인터셉트를 한다
- 왼쪽 수비수로 테오 에르난데스 선수를 추천한다
- 테오 에르난데스는 26살의 전성기에 들어가는 선수이다
- 왼쪽 윙백인데 공격포인트를 7개를 생성하였다
- 키패스를 90분 경기 기준 매경기 1개이상씩 생성할정도로 공격적으로 큰 도움을 준다
- 90기준 매경기 한번 이상의 드리블 시도가 있을정도로 활발한 공격을 도전한다
- 경합을 90분기준 매경기 평규 7번씩 도전하며 승률은 55%이다
- 인터셉트와 블록이 90분기준 매경기 평균 0.6개씩 하며 밸랜스가 있는 선수이다
- 득점지원이 부족한 맨체스터 유나이티드에게는 테오 에르난데스가 좋은 선택지가 될것이다

In [118]:
world_player_midfielder_suggest = spark.sql("""With players as(SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname,pf.player_birth,pf.player_nationality,TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, eps.*,duels_won/duels_total,dribbles_success/dribbles_attempts,game_lineups/game_appearances,passes_total/(game_minutes/90),passes_key/(game_minutes/90),tackles_blocks/(game_minutes/90),tackles_interceptions/(game_minutes/90),duels_total/(game_minutes/90),dribbles_attempts/(game_minutes/90),game_minutes/90 total_games,total_shots/(game_minutes/90),shots_on/(game_minutes/90),goals_assists/(game_minutes/90),passes_key/passes_total,shots_on/total_shots,goals_assists/passes_key,fouls_drawn/(game_minutes/90),fouls_committed/(game_minutes/90),cards_yellow/(game_minutes/90),cards_yellow/fouls_committed,cards_yellowred/(game_minutes/90),cards_red/fouls_committed,total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
FROM england_player_stat eps
JOIN player_df pf ON eps.player_id = pf.player_id
JOIN team_df tf ON eps.team_id = tf.team_id
Union 
SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname,pf.player_birth,pf.player_nationality,TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, sps.*,duels_won/duels_total,dribbles_success/dribbles_attempts,game_lineups/game_appearances,passes_total/(game_minutes/90),passes_key/(game_minutes/90),tackles_blocks/(game_minutes/90),tackles_interceptions/(game_minutes/90),duels_total/(game_minutes/90),dribbles_attempts/(game_minutes/90),game_minutes/90 total_games,total_shots/(game_minutes/90),shots_on/(game_minutes/90),goals_assists/(game_minutes/90),passes_key/passes_total,shots_on/total_shots,goals_assists/passes_key,fouls_drawn/(game_minutes/90),fouls_committed/(game_minutes/90),cards_yellow/(game_minutes/90),cards_yellow/fouls_committed,cards_yellowred/(game_minutes/90),cards_red/fouls_committed,total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
from spain_player_stat sps
JOIN player_df pf ON sps.player_id = pf.player_id
JOIN team_df tf ON sps.team_id = tf.team_id

Union 
SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname,pf.player_birth,pf.player_nationality,TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, ips.*,duels_won/duels_total,dribbles_success/dribbles_attempts,game_lineups/game_appearances,passes_total/(game_minutes/90),passes_key/(game_minutes/90),tackles_blocks/(game_minutes/90),tackles_interceptions/(game_minutes/90),duels_total/(game_minutes/90),dribbles_attempts/(game_minutes/90),game_minutes/90 total_games,total_shots/(game_minutes/90),shots_on/(game_minutes/90),goals_assists/(game_minutes/90),passes_key/passes_total,shots_on/total_shots,goals_assists/passes_key,fouls_drawn/(game_minutes/90),fouls_committed/(game_minutes/90),cards_yellow/(game_minutes/90),cards_yellow/fouls_committed,cards_yellowred/(game_minutes/90),cards_red/fouls_committed,total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
from italy_player_stat ips
JOIN player_df pf ON ips.player_id = pf.player_id
JOIN team_df tf ON ips.team_id = tf.team_id

Union 
SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname,pf.player_birth,pf.player_nationality,TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, gps.*,duels_won/duels_total,dribbles_success/dribbles_attempts,game_lineups/game_appearances,passes_total/(game_minutes/90),passes_key/(game_minutes/90),tackles_blocks/(game_minutes/90),tackles_interceptions/(game_minutes/90),duels_total/(game_minutes/90),dribbles_attempts/(game_minutes/90),game_minutes/90 total_games,total_shots/(game_minutes/90),shots_on/(game_minutes/90),goals_assists/(game_minutes/90),passes_key/passes_total,shots_on/total_shots,goals_assists/passes_key,fouls_drawn/(game_minutes/90),fouls_committed/(game_minutes/90),cards_yellow/(game_minutes/90),cards_yellow/fouls_committed,cards_yellowred/(game_minutes/90),cards_red/fouls_committed,total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
from germany_player_stat gps
JOIN player_df pf ON gps.player_id = pf.player_id
JOIN team_df tf ON gps.team_id = tf.team_id

Union 
SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname, pf.player_birth, pf.player_nationality, TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, fps.*, duels_won/duels_total, dribbles_success/dribbles_attempts, game_lineups/game_appearances, passes_total/(game_minutes/90), passes_key/(game_minutes/90), tackles_blocks/(game_minutes/90), tackles_interceptions/(game_minutes/90), duels_total/(game_minutes/90), dribbles_attempts/(game_minutes/90), game_minutes/90 total_games, total_shots/(game_minutes/90), shots_on/(game_minutes/90), goals_assists/(game_minutes/90), passes_key/passes_total, shots_on/total_shots, goals_assists/passes_key, fouls_drawn/(game_minutes/90), fouls_committed/(game_minutes/90), cards_yellow/(game_minutes/90), cards_yellow/fouls_committed, cards_yellowred/(game_minutes/90), cards_red/fouls_committed, total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
FROM france_player_stat fps
JOIN player_df pf ON fps.player_id = pf.player_id
JOIN team_df tf ON fps.team_id = tf.team_id)
                                          
select * from players
WHERE game_position LIKE 'Midfielder' AND game_appearances > 13 AND tackles_blocks/(game_minutes/90)>0.6 AND tackles_interceptions/(game_minutes/90) >0.6 AND passes_key/(game_minutes/90)>0.5;


""")

In [119]:
world_player_midfielder_suggest.show()

24/03/12 16:04:22 WARN DAGScheduler: Broadcasting large task binary with size 1054.7 KiB


+--------------------+----------------+---------------+------------+------------------+----------+---------+-------+---------+-------------+----------------+------------+------------+-----------+-------------+-----------+------------+------+-------+---------+-----------+--------+-----------+--------------+-------------+-----------+------------+----------+---------------+-------------+--------------+---------------------+-----------+---------+-----------------+----------------+-------------+-----------+---------------+------------+---------------+---------+-----------+----------------+--------------+--------------+-------------+-------------------------+--------------------------------------+---------------------------------+------------------------------------+----------------------------------+--------------------------------------+---------------------------------------------+-----------------------------------+-----------------------------------------+------------------+------------

- 수비형 미드필더 추천으로 달스가르드를 추천한다
- 공격포인트를 3개를 만들어냈다
- 경합을 90분기준 매경기 평균 11번하는데 승률56%를 보여준다
- 키패스도 매경기 평균 1개 이상 보여줄 정도로 패스도 좋다
- 블록과 태클이 90분기준 매경기 평균 0.95개와 0.81을 기록할 정도로 수비에서 좋은 활약을 보여주고있다
- 90분경기기준 매경기 약 1개의 슈팅을 시도할 정도 공격력이 부족하지도않다
- 파울을 46번 범하였는데 4개의 경고 카드를 받으면서 위험지역에서의 파울은 적은편이다
- 분데스리가에서 뛰고있어서 새로운 리그에서 적응기간이 필요하다
- 공격적인 선수가 많은 맨체스터 유나이티드에게 수비적인 선수를 추천한다

In [120]:
world_player_forward_suggest = spark.sql("""With players as(SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname,pf.player_birth,pf.player_nationality,TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, eps.*,duels_won/duels_total,dribbles_success/dribbles_attempts,game_lineups/game_appearances,passes_total/(game_minutes/90),passes_key/(game_minutes/90),tackles_blocks/(game_minutes/90),tackles_interceptions/(game_minutes/90),duels_total/(game_minutes/90),dribbles_attempts/(game_minutes/90),game_minutes/90 total_games,total_shots/(game_minutes/90),shots_on/(game_minutes/90),goals_assists/(game_minutes/90),passes_key/passes_total,shots_on/total_shots,goals_assists/passes_key,fouls_drawn/(game_minutes/90),fouls_committed/(game_minutes/90),cards_yellow/(game_minutes/90),cards_yellow/fouls_committed,cards_yellowred/(game_minutes/90),cards_red/fouls_committed,total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
FROM england_player_stat eps
JOIN player_df pf ON eps.player_id = pf.player_id
JOIN team_df tf ON eps.team_id = tf.team_id
Union 
SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname,pf.player_birth,pf.player_nationality,TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, sps.*,duels_won/duels_total,dribbles_success/dribbles_attempts,game_lineups/game_appearances,passes_total/(game_minutes/90),passes_key/(game_minutes/90),tackles_blocks/(game_minutes/90),tackles_interceptions/(game_minutes/90),duels_total/(game_minutes/90),dribbles_attempts/(game_minutes/90),game_minutes/90 total_games,total_shots/(game_minutes/90),shots_on/(game_minutes/90),goals_assists/(game_minutes/90),passes_key/passes_total,shots_on/total_shots,goals_assists/passes_key,fouls_drawn/(game_minutes/90),fouls_committed/(game_minutes/90),cards_yellow/(game_minutes/90),cards_yellow/fouls_committed,cards_yellowred/(game_minutes/90),cards_red/fouls_committed,total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
from spain_player_stat sps
JOIN player_df pf ON sps.player_id = pf.player_id
JOIN team_df tf ON sps.team_id = tf.team_id

Union 
SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname,pf.player_birth,pf.player_nationality,TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, ips.*,duels_won/duels_total,dribbles_success/dribbles_attempts,game_lineups/game_appearances,passes_total/(game_minutes/90),passes_key/(game_minutes/90),tackles_blocks/(game_minutes/90),tackles_interceptions/(game_minutes/90),duels_total/(game_minutes/90),dribbles_attempts/(game_minutes/90),game_minutes/90 total_games,total_shots/(game_minutes/90),shots_on/(game_minutes/90),goals_assists/(game_minutes/90),passes_key/passes_total,shots_on/total_shots,goals_assists/passes_key,fouls_drawn/(game_minutes/90),fouls_committed/(game_minutes/90),cards_yellow/(game_minutes/90),cards_yellow/fouls_committed,cards_yellowred/(game_minutes/90),cards_red/fouls_committed,total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
from italy_player_stat ips
JOIN player_df pf ON ips.player_id = pf.player_id
JOIN team_df tf ON ips.team_id = tf.team_id

Union 
SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname,pf.player_birth,pf.player_nationality,TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, gps.*,duels_won/duels_total,dribbles_success/dribbles_attempts,game_lineups/game_appearances,passes_total/(game_minutes/90),passes_key/(game_minutes/90),tackles_blocks/(game_minutes/90),tackles_interceptions/(game_minutes/90),duels_total/(game_minutes/90),dribbles_attempts/(game_minutes/90),game_minutes/90 total_games,total_shots/(game_minutes/90),shots_on/(game_minutes/90),goals_assists/(game_minutes/90),passes_key/passes_total,shots_on/total_shots,goals_assists/passes_key,fouls_drawn/(game_minutes/90),fouls_committed/(game_minutes/90),cards_yellow/(game_minutes/90),cards_yellow/fouls_committed,cards_yellowred/(game_minutes/90),cards_red/fouls_committed,total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
from germany_player_stat gps
JOIN player_df pf ON gps.player_id = pf.player_id
JOIN team_df tf ON gps.team_id = tf.team_id

Union 
SELECT DISTINCT tf.team_name, pf.player_firstname, pf.player_lastname, pf.player_birth, pf.player_nationality, TIMESTAMPDIFF(YEAR, pf.player_birth, CURDATE()) AS player_age, fps.*, duels_won/duels_total, dribbles_success/dribbles_attempts, game_lineups/game_appearances, passes_total/(game_minutes/90), passes_key/(game_minutes/90), tackles_blocks/(game_minutes/90), tackles_interceptions/(game_minutes/90), duels_total/(game_minutes/90), dribbles_attempts/(game_minutes/90), game_minutes/90 total_games, total_shots/(game_minutes/90), shots_on/(game_minutes/90), goals_assists/(game_minutes/90), passes_key/passes_total, shots_on/total_shots, goals_assists/passes_key, fouls_drawn/(game_minutes/90), fouls_committed/(game_minutes/90), cards_yellow/(game_minutes/90), cards_yellow/fouls_committed, cards_yellowred/(game_minutes/90), cards_red/fouls_committed, total_goals/shots_on, total_goals-penalty_scored as field_score, total_goals+goals_assists as total_attack_points
FROM france_player_stat fps
JOIN player_df pf ON fps.player_id = pf.player_id
JOIN team_df tf ON fps.team_id = tf.team_id)
                                          
select * from players
WHERE game_position LIKE 'Attacker' AND game_appearances > 13 AND (total_shots / (game_minutes / 90))>2 AND dribbles_attempts/(game_minutes/90)>3 AND  total_attack_points>10;


""")

In [121]:
world_player_forward_suggest.show()

24/03/12 16:04:26 WARN DAGScheduler: Broadcasting large task binary with size 1050.3 KiB


+-------------------+--------------------+--------------------+------------+------------------+----------+---------+-------+---------+-------------+----------------+------------+------------+-----------+-------------+-----------+------------+------+-------+---------+-----------+--------+-----------+--------------+-------------+-----------+------------+----------+---------------+-------------+--------------+---------------------+-----------+---------+-----------------+----------------+-------------+-----------+---------------+------------+---------------+---------+-----------+----------------+--------------+--------------+-------------+-------------------------+--------------------------------------+---------------------------------+------------------------------------+----------------------------------+--------------------------------------+---------------------------------------------+-----------------------------------+-----------------------------------------+------------------+----

- 쿠냐 선수 영입을 추천한다
- 24살의 어린선수이고 경기도 꾸준히 뛰고 있다
- 공격포인트를 15개를 할정도로 공격적 측면에서 좋은 활약을 보여주고있다
- 드리블을 매격기 평균 4개를 할정도로 도전적이면서 성공률이 53%로 공격적인 활로가 될수 있다
- 키패스도 매경기 평균 1개씩해주면서 질 좋은 패스를 주고있다
- 경합도 매경기 평균 13개씩 해주면서 수비적으로도 도움을 많이 준다
- 슈팅도 매경기 평균 2개씩 시도하며 유효슈팅은 매경기 평균 1개이상씩 만든다
- 파울도 평균 1개 이상씩 얻어 주면서 현소속팀인 울브스의 핵심선수이다
- 공격수로도 뛰고 윙어로도 뛸수있어 다양하게 사용할수있는 옵션이 된다
- 같은 리그에서 뛰고있기때문에 적응기간이 더 짧다
